common/src/test/java/tachyon/security/authentication/AuthenticationFactoryTest.java

```diff
@@ -22,49 +43,387 @@
+    authType = new AuthenticationFactory(mTachyonConf).getAuthType();
+  }
+
+  /**
+   * In NOSASL mode, the TTransport used should be the same as Tachyon original code.
+   */
+  @Test
+  public void nosaslAuthenticationTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "NOSASL");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // create client and connect to server
+    TTransport client = createClient(mTachyonConf, null, null);
+    client.open();
+    Assert.assertTrue(client.isOpen());
+
+    // clean up
+    client.close();
+    mServer.stop();
+  }
+
+  /**
+   * In SIMPLE mode, the TTransport mechanism is PLAIN. When server authenticate the connected
+   * client user, it use {@link tachyon.security.authentication.SimpleAuthenticationProviderImpl}.
+   */
+  @Test
+  public void simpleAuthenticationTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // when connecting, authentication happens. It is a no-op in Simple mode.
+    TTransport client = createClient(mTachyonConf, "anyone", "whatever");
+    client.open();
+    Assert.assertTrue(client.isOpen());
+
+    // clean up
+    client.close();
+    mServer.stop();
+  }
+
+  /**
+   * In SIMPLE mode, if client's username is null, an exception should be thrown in client side.
+   */
+  @Test
+  public void simpleAuthenticationNullUserTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // check case that user is null
+    mThrown.expect(SaslException.class);
+    mThrown.expectMessage("PLAIN: authorization ID and password must be specified");
+    TTransport client = createClient(mTachyonConf, null, "whatever");
+  }
+
+  /**
+   * In SIMPLE mode, if client's password is null, an exception should be thrown in client side.
+   */
+  @Test
+  public void simpleAuthenticationNullPasswordTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // check case that password is null
+    mThrown.expect(SaslException.class);
+    mThrown.expectMessage("PLAIN: authorization ID and password must be specified");
+    TTransport client = createClient(mTachyonConf, "anyone", null);
+  }
+
+  /**
+   * In SIMPLE mode, if client's username is empty, an exception should be thrown in server side.
+   */
+  @Test
+  public void simpleAuthenticationEmptyUserTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // check case that user is empty
+    mThrown.expect(TTransportException.class);
+    mThrown.expectMessage("Peer indicated failure: Plain authentication failed: No authentication"
+        + " identity provided");
+    TTransport client = createClient(mTachyonConf, "", "whatever");
+    try {
```

Comment:
why we `try` here? If we are not expecting any exception thrown, we shall not have `try` here.